zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt

```diff
@@ -60,29 +61,46 @@ open class PortaalTaakEventListener(
     @RunWithoutAuthorization
     @EventListener(NotificatiesApiNotificationReceivedEvent::class)
     open fun processCompletePortaalTaakEvent(event: NotificatiesApiNotificationReceivedEvent) {
-        val objectType = event.kenmerken["objectType"]
+        logger.debug { "Received Notificaties API event, checking if it fits criteria to complete a portaaltaak" }
 
+        val objectType = event.kenmerken["objectType"]
         if (!event.kanaal.equals("objecten", ignoreCase = true) ||
             !event.actie.equals("update", ignoreCase = true) ||
             objectType == null
         ) {
+            logger.debug { "Notificaties API event does not match criteria for completing a portaaltaak. Ignoring." }
             return
         }
 
+        val objectTypeId = objectType.substringAfterLast("/")
+
         val objectManagement =
-            objectManagementService.findByObjectTypeId(objectType.substringAfterLast("/")) ?: return
+            objectManagementService.findByObjectTypeId(objectTypeId)
+                ?: run {
+                    logger.warn { "Object management not found for object type id '$objectTypeId'" }
+                    return
+                }
 
-        pluginService.findPluginConfiguration(PortaaltaakPlugin::class.java) { properties: JsonNode
-            ->
+        pluginService.findPluginConfiguration(PortaaltaakPlugin::class.java) { properties: JsonNode ->
             properties.get("objectManagementConfigurationId").textValue().equals(objectManagement.id.toString())
         }?.let {
+            logger.debug { "Completing portaaltaak using plugin configuration with id '${it.id}'" }
 
-            val taakObject: TaakObject =
-                objectMapper.convertValue(getPortaalTaakObjectData(objectManagement, event))
+            val taakObject: TaakObject = objectMapper.convertValue(getPortaalTaakObjectData(objectManagement, event))
             when (taakObject.status) {
                 TaakStatus.INGEDIEND -> {
-                    val task = taskService.findTaskById(taakObject.verwerkerTaakId) ?: return
-                    val receiveData = getReceiveDataActionProperty(task, it.id.id) ?: return
+                    logger.debug { "Processing task with status 'ingediend' and verwerker task id '${taakObject.verwerkerTaakId}'" }
+                    val task = taskService.findTaskById(taakObject.verwerkerTaakId)
+                        ?: run {
+                            logger.warn { "Task not found with verwerker task id '${taakObject.verwerkerTaakId}'" }
+                            return
+                        }
+
+                    val receiveData = getReceiveDataActionProperty(task, it.id.id)
+                        ?: run {
+                            logger.warn { "Receive data not found for task with id '${task.id}'" }
```

Comment:
I think you're logging this twice, once in the `getReceiveDataActionProperty` method and here. Ideally logging messages are unique. My suggestion would be to change one of the logging messages so it gives a bit more information.